addons/website_sale/controllers/main.py

```diff
@@ -254,15 +254,14 @@ def shop(self, page=0, category=None, search='', ppg=False, **post):
                 parent_category_ids.append(current_category.parent_id.id)
                 current_category = current_category.parent_id
 
-        product_count = Product.search_count(domain)
+        search_product = Product.search(domain)
```

Comment:
Can we use order by here, to allow to make the substract (limit / offset) by hand later L260 ?
It will make the request only once.

Very Surprised to see that provide 160 000 ids is faster than processing the request a second time.
Because if i'm not wrong, the ids will be splitted by slot and so make a lot of queries.

But why use private attributes ._ids instead of .ids ?

Reply:
Sure I didn't thought about it. The code was updated.
Well I was quite surprised but the count and the other search was almost as time consuming as the search without restrictions (see pgbadger and pyflame files in ticket log note).

Note that on the original code all products were fetched any way, so it looks smarter to me to do it this way.

I wasn't aware that `.ids` exists, my bad!